server/routes/users.js

```diff
@@ -93,598 +93,596 @@ router.get('/login', redirectProfile, (req, res) => {
 // @tested 	Yes
 // TODO: add conditions to check userRole and limit 'createWishCard' access to 'partners' only
 router.get('/profile', redirectLogin, async (req, res) => {
-	try {
-		const { user } = req.session;
-		if (user.userRole === 'partner') {
-			const agency = await AgencyRepository.getAgencyByUserId(user._id);
-
-			// If user hadn't filled out agency info, redirect them to form
-			if (!agency) {
-				return res.status(200).render('agency');
-			}
-			const wishCards = await WishCardRepository.getWishCardByAgencyId(agency._id);
-			const wishCardsLength = wishCards.length;
-			res.status(200).render('profile', { wishCardsLength });
-		} else {
-			res.status(200).render('profile');
-		}
-	} catch (err) {
-		return handleError(res, 400, err);
-	}
+  try {
+    const { user } = req.session;
+    if (user.userRole === 'partner') {
+      const agency = await AgencyRepository.getAgencyByUserId(user._id);
+
+      // If user hadn't filled out agency info, redirect them to form
+      if (!agency) {
+        return res.status(200).render('agency');
+      }
+      const wishCards = await WishCardRepository.getWishCardByAgencyId(agency._id);
+      const wishCardsLength = wishCards.length;
+      res.status(200).render('profile', { wishCardsLength })
```

Comment:
i would put wishcardslength in the block before with let wishcardsLength and the n just give 			res.status(200).render('profile', { wishCardsLength }); back. saves an else

Reply:
old server, that's from merging the master